### ISSUE TITLE
Allow LowCardinality for UUID field without allow_suspicious_low_cardinality_types=1

### ISSUE THREAD

**Issue by socketpair** [ClickHouse/ClickHouse#56818]
#4965

issues message about inefficiency if `LowCardinality` from non-strings (although documentation https://clickhouse.com/docs/en/sql-reference/data-types/lowcardinality says `DateTime` is OK).

In my project, I don't need low cardinality `DateTime` columns, but for `UUID` columns, optimization can be applied.

This shows that `LowCardinality (UUID)` makes sense in terms of storage size at about **3 times better compression** (if we calculate disk storage size without of column `id`).
https://fiddle.clickhouse.com/d3d8ed3c-e444-4f04-a42a-7e14d28750bb

So, I propose these changes:
1. Allow to use `LowCardinality (UUID)` without settings `allow_suspicious_low_cardinality_types=1`
2. Allow `Datetime` types which more than 8 bytes long in storage.
3. Possibly issue warning back on short `FixedString`s. But I am unsure here because as I think, string operations can be applied first to internal dict data and after that these ops will not be required during raw data table scan. So it might give optimizations in some cases even on short strings.

**Comment by socketpair** [comment]
In telegram, people wonder what is the real case ? Actually, this case is absolutely real. I have some external system that send me a stream of events where the subject of every event is an ID of remote object in form of UUID.

**Comment by Slach** [comment]
> Allow to use LowCardinality (UUID) without settings allow_suspicious_low_cardinality_types=1

It will inefficient when UUID will high cardinality values and will broke behavior for other users which not understand what usually UUID means

using LowCardinality(UUID) shall be a decision from user side 
just add setting in your default profile and live happy

for example if you have high cardinality values you will lose
https://fiddle.clickhouse.com/53898c5a-76a3-4bde-9ac2-47c3cfe82e83

**Comment by den-crane** [comment]
>Allow Datetime types which more than 8 bytes long in storage.

But there is no Datetime types which are more than 8 bytes

**Comment by socketpair** [comment]
@Slach yes, using LowCardinality on data which actually high cardinality is a stupid idea, the same for String columns. So, in your case it is expected. Exactly the same as if it were String column.

**Comment by Taeknology** [comment]
Looks like proposal #1 (`LowCardinality(UUID)` without `allow_suspicious_low_cardinality_types=1`) was already implemented in #73826 ("Allow `LowCardinality(UUID)` by default"), merged 2024-12-27.

The current check in `src/Interpreters/parseColumnsListForTableFunction.cpp` explicitly exempts UUID:

```cpp
/// It is allowed having LowCardinality(UUID) because often times UUIDs are highly repetitive in tables,
/// and their relatively large size provides opportunity for better performance.
if (!isStringOrFixedString(unwrapped) && !isUUID(unwrapped))
    throw Exception(...);
```

Proposal #2 (DateTime > 8 bytes) doesn't apply — DateTime/DateTime64 are both 8 bytes, as @den-crane noted. Proposal #3 (warning on short FixedString) the author left as uncertain.

If maintainers agree the actionable part is done, this issue could probably be closed.
